pkg/agent/deployer/generic/generic.go

```diff
@@ -233,13 +233,14 @@ func (deployer *Deployer) ControllerHasStarted(gvk schema.GroupVersionKind) bool
 	}
 }
 
-func (deployer *Deployer) AddController(gvk schema.GroupVersionKind, controller *resourcecontroller.Controller) {
+func (deployer *Deployer) AddController(gvk schema.GroupVersionKind, controller *resourcecontroller.Controller) bool {
 	deployer.lock.Lock()
 	defer deployer.lock.Unlock()
 	if _, ok := deployer.rsControllers[gvk]; ok {
-		return
+		return false
```

Comment:
`False` is misleading. For a single GVK, we have no idea whether we've successfully added a matching controller.

Reply:
If false is returned, it can be determined that a resource has triggered to start matching controller

Reply:
The main issue here is that the judgment of 152 line above and the startup of controller here are not atomic operations